test/cat.jl

```diff
@@ -143,35 +142,34 @@ module TestCat
     end
 
     @testset "vcat mixed coltypes" begin
-        drf = CategoricalArrays.DefaultRefType
         dt = vcat(DataTable([[1]], [:x]), DataTable([[1.0]], [:x]))
         @test dt == DataTable([[1.0, 1.0]], [:x])
         @test typeof.(dt.columns) == [Vector{Float64}]
         dt = vcat(DataTable([[1]], [:x]), DataTable([["1"]], [:x]))
         @test dt == DataTable([[1, "1"]], [:x])
         @test typeof.(dt.columns) == [Vector{Any}]
-        dt = vcat(DataTable([NullableArray([1])], [:x]), DataTable([[1]], [:x]))
-        @test dt == DataTable([NullableArray([1, 1])], [:x])
-        @test typeof.(dt.columns) == [NullableVector{Int}]
+        dt = vcat(DataTable([Union{Null, Int}[1]], [:x]), DataTable([[1]], [:x]))
+        @test dt == DataTable([[1, 1]], [:x])
+        @test typeof.(dt.columns) == [Vector{Union{Null, Int}}]
         dt = vcat(DataTable([CategoricalArray([1])], [:x]), DataTable([[1]], [:x]))
         @test dt == DataTable([CategoricalArray([1, 1])], [:x])
-        @test typeof.(dt.columns) == [CategoricalVector{Int, drf}]
+        @test typeof(dt[:x]) <: CategoricalVector{Int}
         dt = vcat(DataTable([CategoricalArray([1])], [:x]),
-                  DataTable([NullableArray([1])], [:x]))
-        @test dt == DataTable([NullableCategoricalArray([1, 1])], [:x])
-        @test typeof.(dt.columns) == [NullableCategoricalVector{Int, drf}]
+                  DataTable([Union{Null, Int}[1]], [:x]))
+        @test dt == DataTable([CategoricalArray([1, 1])], [:x])
+        @test typeof(dt[:x]) <: CategoricalVector{Union{Int, Null}}
         dt = vcat(DataTable([CategoricalArray([1])], [:x]),
-                  DataTable([NullableCategoricalArray([1])], [:x]))
-        @test dt == DataTable([NullableCategoricalArray([1, 1])], [:x])
-        @test typeof.(dt.columns) == [NullableCategoricalVector{Int, drf}]
-        dt = vcat(DataTable([NullableArray([1])], [:x]),
-                  DataTable([NullableArray(["1"])], [:x]))
-        @test dt == DataTable([NullableArray([1, "1"])], [:x])
-        @test typeof.(dt.columns) == [NullableVector{Any}]
+                  DataTable([CategoricalArray{Union{Int, Null}}([1])], [:x]))
+        @test dt == DataTable([CategoricalArray{Union{Int, Null}}([1, 1])], [:x])
+        @test typeof(dt[:x]) <: CategoricalVector{Union{Int, Null}}
+        dt = vcat(DataTable([[1]], [:x]),
```

Comment:
This one was nullable too. OTOH, no need to convert to `CategoricalArray` above when checking for `==`.